### ISSUE TITLE
SearchBar does not properly support RTL

### ISSUE THREAD

**Issue by alanbsmith** [Workday/canvas-kit#1226]
## 🐛 Bug Report

SearchBar's search Icon and SearchComboBox do not have proper bidirectional support. 

They use [Emotion's `styled` function](https://github.com/Workday/canvas-kit/blob/master/modules/labs-react/header/lib/parts/SearchBar.tsx#L2) instead of our `styled` function in `@workday/canvas-kit-react/common`.

**This is something we'll need to fix in 4.x and 5.x**

## To Reproduce

[View this sandbox](https://codesandbox.io/s/seachbar-v5-1-0-rtl-test-v3fu8?file=/src/App.tsx)

## Expected Behavior

 and 
When the theme's direction is set to `ContentDirection.RTL`:

- the search icon should switch to the right
- `autoCompleteItems` in the SearchCombobox should have right-aligned text

## Actual Results

The icon position and menu item text alignment remains the same.

## Browser (if applicable)

n/a

## Link to repl or repo (highly encouraged)

[Sandbox]https://codesandbox.io/s/seachbar-v5-1-0-rtl-test-v3fu8?file=/src/App.tsx)

### Error Output

![Screen Shot 2021-08-16 at 23 45 56](https://user-images.githubusercontent.com/4818182/129670245-54611753-4922-4cfb-bea8-37991244f5bc.png)

## Acceptance Criteria

- [ ] Fix SearchBar styles so that it has proper bidirectional support.

**Comment by vibdev** [comment]
This looks like it is fixed 👍

**Comment by vibdev** [comment]
This looks like it is fixed 👍

**Comment by myvuuu** [comment]
Closing out!
